sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/models/AttestationOptions.java

```diff
@@ -0,0 +1,157 @@
+     */
+    byte[] getRunTimeData();
+
+    /**
+     * When the attestation service receives the RunTime Data, it should interpret the data as
+     * binary, and return the data in the {@link AttestationResult#getEnclaveHeldData()} property
+     * @return the AttestationOptions object.
+     */
+    AttestationOptions interpretRunTimeDataAsBinary();
+
+    /**
+     * When the attestation service receives the RunTime Data, it should interpret the data as
+     * binary, and return the data in the {@link AttestationResult#getRuntimeClaims()} property
+     * @return the AttestationOptions object.
+     */
+    AttestationOptions interpretRunTimeDataAsJson();
+
+    /**
+     * Set the initTimeData property: Base64Url encoded "InitTime data". The MAA will verify that the init data was
+     * known to the enclave. Note that InitTimeData is invalid for CoffeeLake processors.
+     *
+     * @param initTimeData the initTimeData value to set.
+     * @return the AttestOpenEnclaveRequest object itself.
+     */
+    AttestationOptions setInitTimeData(byte[] initTimeData);
+
+    /**
+     * Retrieves the InitTimeData property to be sent to the service.
+     * @return The InitTimeData value set by {@link AttestationOptions#setInitTimeData}
+     */
+    byte[] getInitTimeData();
+
+    /**
+     * When the attestation service receives the InitTime Data, it should interpret the data as
+     * binary, and return the data in the {@link AttestationResult#getInittimeClaims()} property
+     * @return the AttestationOptions object.
+     */
+    AttestationOptions interpretInitTimeDataAsBinary();
+
+    /**
+     * When the attestation service receives the InitTime Data, it should interpret the data as
+     * binary, and return the data in the {@link AttestationResult#getInittimeClaims()} property
+     * @return the AttestationOptions object.
+     */
+    AttestationOptions interpretInitTimeDataAsJson();
+
+    /**
+     * Set the draftPolicyForAttestation property: Attest against the provided draft policy. Note that the resulting
+     * token cannot be validated.
+
+     * Note that the resulting token cannot be validated.
+     *
+     * @param draftPolicyForAttestation the draftPolicyForAttestation value to set.
+     * @return the AttestOpenEnclaveRequest object itself.
+     */
+    AttestationOptions setDraftPolicyForAttestation(String draftPolicyForAttestation);
+
+    /**
+     * Gets the draftPolicyForAttestation property which is used to attest against the draft policy.
+     *
+     * Note that the resulting token cannot be validated.
+     *
+     * @return The draft policy if set.
+     */
+    String getDraftPolicyForAttestation();
+
+    /**
+     * Validates the instance.
+     *
+     * If the caller has called setRunTimeData without calling interpretRunTimeDataAsBinary or
+     * interpretRunTimeDataAsJson, this will throw an exception, similarly if you call setInitTimeData
```

Comment:
You could get rid of the need to validate by encapsultating this in an object that has both Mode and Data as required fields.  I understand you prefer this but it can confuse end users the first time they use it.

Reply:
The purpose of the `setRuntimeData` and `setRuntimeJson` APIs that you had challenges with was to avoid the need for validation. Moving this into a separate AttestationData object doesn't help the problem. You go from:

```java
options
    .setRunTimeData(data)
    .interpretRunTimeDataAsBinary();
```

to:
```
options
    .setRunTimeData(new AttestationData()
        .setData(data)
        .interpretDataAsBinary());
```

Either way, there are two fields which MUST to be set, one for the data and one for how it's interpreted.